From 64fee2d5ee42688a813a33b0d2f047f3dcc6b56f Mon Sep 17 00:00:00 2001
From: John Weber <rjohn...@gmail.com>
Date: Thu, 28 Feb 2013 13:23:26 -0600
Subject: [PATCH 053/225] Wandboard : Fix SDHC platform data

Add 'always_present = 1' setting to the platform data for the ESDHC
controller which is dedicated to the Broadcom wifi device (BCM4329).
Without this change, the kernel has problems downloading firmware to
the chip.

Made some other modifications to eliminate annoying boot messages,
and the addition of the .cd_type to the data which tells the driver
what type of CD signal to expect (i.e. for permanent on-board chips
this is different from removable devices).

Signed-off-by: John Weber <rjohn...@gmail.com>
---
 arch/arm/mach-mx6/board-wand.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mach-mx6/board-wand.c b/arch/arm/mach-mx6/board-wand.c
index e01b54f..276c79d 100644
--- a/arch/arm/mach-mx6/board-wand.c
+++ b/arch/arm/mach-mx6/board-wand.c
@@ -190,12 +190,18 @@ static const struct esdhc_platform_data wand_sd_data[3] = {
 		.wp_gpio		=-EINVAL,
 		.keep_power_at_suspend	= 1,
 	        .support_8bit		= 0,
+		.delay_line		= 0,
 		.platform_pad_change	= wand_sd_speed_change,
+                .cd_type                = ESDHC_CD_CONTROLLER,
 	}, {
 		.cd_gpio		=-EINVAL,
 		.wp_gpio		=-EINVAL,
 		.keep_power_at_suspend	= 1,
+		.support_8bit		= 0,
+		.delay_line		= 0,
 		.platform_pad_change	= wand_sd_speed_change,
+                .always_present		= 1,
+                .cd_type                = ESDHC_CD_PERMANENT,
 	}, {
 		.cd_gpio		= WAND_SD3_CD,
 		.wp_gpio		= WAND_SD3_WP,
@@ -203,6 +209,7 @@ static const struct esdhc_platform_data wand_sd_data[3] = {
 		.support_8bit		= 0,
 		.delay_line		= 0,
 		.platform_pad_change	= wand_sd_speed_change,
+                .cd_type                = ESDHC_CD_CONTROLLER,
 	}
 };
 
-- 
1.7.10.4

